### DISCUSSION TITLE
jupyterlab-myst contribution in the frame of a JupyterLab CZI grant

### DISCUSSION THREAD

**Discussion by fcollonval** [executablebooks/meta#748] [General]
In the JupyterLab [user survey](https://github.com/jupyterlab/team-compass/issues/121#issuecomment-782913362), one of the key points highlighted is to support documenting research and creating content. Your awesome work is definitely addressing that. And in particular the [jupyterlab-myst](https://github.com/executablebooks/jupyterlab-myst) extension sounds like the best vessel to mature the integration in JupyterLab. In the proposal for a [JupyterLab CZI grant](https://github.com/jupyterlab/team-compass/pull/140), we (QuantStack) are planning to work on that feature. So the question is: would it be ok to contribute to `jupyterlab-myst` to move forward with that feature in the frame of the CZI grant (if it gets accepted)?

**Comment by choldgraf** [comment]
I think it would be impactful for both the lab and book ecosystem, and I like the idea of building connections across these communities.

I'd also like to make sure that @agoose77 is on board with the idea as he has done much of the work in this repository thus far. I'd also suggest interfacing with @chrisjsewell and @rowanc1 as they've been pushing on a lot of the Myst JavaScript work that is being used under the hood by JupyterLab-myst

**Comment by agoose77** [comment]
Well, this is _very_ exciting. On my end, I'm completely on board with any external involvement. Whilst for the most part `juptyerlab-myst` leverages the JS Myst ecosystem from @chrisjsewell, @rowanc1 et al (there are definitely more contributors!), there are a few features that would particularly benefit from additional support (particularly JLab specific). Off the top of my head, I'm thinking of:
- Kernel execution in markdown (http://github.com/agoose77/jupyterlab-imarkdown is not very pretty to look at right now) for the `eval` directive/roles
- Citations (@krassowski has [jupyterlab-citation-manager](https://github.com/krassowski/jupyterlab-citation-manager) which would be exciting to integrate with JB)

I'd personally welcome any and all contributions.

**Comment by fcollonval** [comment]
Thanks both for the replies and especially @agoose77 for pointing out concrete features that will be great additions.

**Reply by choldgraf** [reply]
I'd +1 the idea of using resources to help push forward the "inline execution" story. I think that'll require some prototyping to play around with what the structure and syntax should look like (@chrisjsewell recently added one such example in Myst-nb). It would be great to brainstorm how the mdast and mystjs work could be brought into the jupyterlab world as well.

One thing I should note though, is that the executable books project is not technically a part of the jupyter project right now. It is "just a grant-funded group that does their work on GitHub". We do not have a formal governance model nor a code of conduct process, and this might be a problem for CZI.

Some discussion of how to implement this is here: https://github.com/executablebooks/meta/issues/493#issue-1004897290 though we have not yet begun implementing that work. The grant that funds the current group of people developing the project will run out towards the end of the year I believe.
